app/models/form.rb

```diff
@@ -14,12 +14,19 @@ class Form < ActiveRecord::Base
   has_paper_trail on: [:update, :destroy]
   has_many :form_elements, -> { order(:position) }
 
+  after_touch do
+    [Plugins::Petition, Plugins::Fundraiser].each do |plugin_type|
+      plugin_type.where(form_id: id).each do |plugin|
+        plugin.page.touch
+      end
+    end
+  end
```

Comment:
FIXME - This spits in the face of the open closed principle. 

Possible solutions:
 - `Form#belongs_to :plugin (polymorphic)`
 - Have form using plugins register with `Form` class on code initialisation.